### PR TITLE
reduce startup time

### LINKED ISSUE #1541: reduce startup time of sermant-agent
### What would you like to be enhanced?

When mounting sermant-agent to start the application, the necessary execution steps and bytecode processing will take a certain amount of time. A mechanism can be introduced to pre-enhance or pre-filter before officially starting the application, so that most of the time can be saved when starting sermant-agent and the host application.

### Why is this needed?

To start sermant-agent and host application faster

### PR DESCRIPTION
**Pull request by lilai23**
**What type of PR is this?**

Enhancement

**What this PR does / why we need it?**

1. A pre-filter mechanism has been added. 
(1) First, before starting the application in the production environment,  let the host application mount sermant-agent in the test environment or pre-production environment and run it once and turn on the pre-filter switch. This will generate a txt file containing the class names that do not need to be matched and enhanced in the host application.
 (2) When the production environment is officially running, sermant-agent will obtain the class name file in step 1 and ignore them in the matching filter of bytecode enhancement to save startup time.
Since the classes that need to be enhanced only occupy a very small part, this method can filter out most of the classes and save a lot of matching time.

2. modify some log print logic and optimize some process logic of loading agent

**Which issue(s) this PR fixes？**

Fixes #1541

**Does this PR introduce a user-facing change?**

No

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/sermant-io/Sermant/issues) related with this PR before you start working on it.
- [x] Make sure you have squashed your change to one single commit.
- [ ] GitHub Actions works fine in this PR.

### CODE REVIEW

File: sermant-agentcore/sermant-agentcore-premain/src/main/java/io/sermant/premain/utils/LoggerUtils.java [hunk 1]
```diff
@@ -58,4 +60,20 @@ public String format(LogRecord record) {
...
+
+    /**
+     * record stack trace
+     *
+     * @param throwable throwable
+     * @return record string
+     */
+    public static String recordStackTrace(Throwable throwable) {
+        StringBuilder record = new StringBuilder(
+                throwable + System.lineSeparator());
```
Comment (hanbingleixue) [inline sermant-agentcore/sermant-agentcore-premain/src/main/java/io/sermant/premain/utils/LoggerUtils.java]: StringBuilder record = new StringBuilder(throwable);
record .append(System.lineSeparator());

Comment (lilai23) [reply sermant-agentcore/sermant-agentcore-premain/src/main/java/io/sermant/premain/utils/LoggerUtils.java]: done

File: sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/config/utils/ConfigFieldUtil.java [hunk 2]
```diff
@@ -124,11 +123,11 @@ private static void forceSet(Object obj, Field field, Object value) {
             field.set(obj, value);
             modifiersField.setInt(field, modifiers);
         } catch (NoSuchFieldException e) {
-            LOGGER.log(Level.WARNING, String.format(Locale.ROOT,
-                    "Missing modifiers field [%s] of [%s]. ", field.getName(), field.getDeclaringClass().getName()));
+            LOGGER.log(Level.WARNING, "Missing modifiers field [{0}] of [{1}]. ",
```
Comment (daizhenyu) [inline sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/config/utils/ConfigFieldUtil.java:126]: [{0}] or [%s] should be consistent

Comment (lilai23) [reply sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/config/utils/ConfigFieldUtil.java:126]: done

**Review by hanbingleixue — APPROVED** [review]

**Review by daizhenyu — APPROVED** [review]

**Pull request merged**